src/ol/render/canvas/style.js

```diff
@@ -298,11 +298,16 @@ export function buildStyle(flatStyle, context) {
  * @return {FillEvaluator?} A function that evaluates to a fill.
  */
 function buildFill(flatStyle, prefix, context) {
-  const evaluateColor = colorLikeEvaluator(
-    flatStyle,
-    prefix + 'fill-color',
-    context
-  );
+  let evaluateColor;
+  if ('fill-pattern-src' in flatStyle) {
+    evaluateColor = patternEvaluator(flatStyle, 'fill-', context);
```

Comment:
I think this condition is also met when the text style is evaluated in 

https://github.com/openlayers/openlayers/blob/f757d988bea49cbd3e1296510f2e48a10d8fdb5d/src/ol/render/canvas/style.js#L454

This might be the cause for my issue reported in https://github.com/openlayers/openlayers/pull/15318#issuecomment-1810244483